evmrpc/worker_pool_test.go

```diff
@@ -361,3 +361,237 @@ func TestWorkerPoolOrderingNotGuaranteed(t *testing.T) {
 		}
 	}
 }
+
+// Test InitGlobalWorkerPool with custom configuration
+func TestInitGlobalWorkerPool(t *testing.T) {
+	// Reset global pool for this test
+	// Note: In production this should only be called once at startup
+
+	tests := []struct {
+		name              string
+		workerPoolSize    int
+		workerQueueSize   int
+		expectedWorkers   int
+		expectedQueueSize int
+	}{
+		{
+			name:              "custom values",
+			workerPoolSize:    10,
+			workerQueueSize:   500,
+			expectedWorkers:   10,
+			expectedQueueSize: 500,
+		},
+		{
+			name:              "zero values use defaults",
+			workerPoolSize:    0,
+			workerQueueSize:   0,
+			expectedWorkers:   -1, // Will be runtime.NumCPU() * 2, check > 0
+			expectedQueueSize: DefaultWorkerQueueSize,
+		},
+		{
+			name:              "only worker size specified",
+			workerPoolSize:    20,
+			workerQueueSize:   0,
+			expectedWorkers:   20,
+			expectedQueueSize: DefaultWorkerQueueSize,
+		},
+		{
+			name:              "only queue size specified",
+			workerPoolSize:    0,
+			workerQueueSize:   1000,
+			expectedWorkers:   -1, // Will be runtime.NumCPU() * 2
+			expectedQueueSize: 1000,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Create a new worker pool for each test (not using global)
+			// NewWorkerPool will apply defaults if values are <= 0
+			wp := NewWorkerPool(tt.workerPoolSize, tt.workerQueueSize)
+			wp.Start()
+			defer wp.Close()
+
+			// Apply defaults if needed (simulating InitGlobalWorkerPool logic)
+			expectedWorkers := tt.expectedWorkers
+			if expectedWorkers <= 0 {
+				expectedWorkers = 2 // At least 2 workers expected on any system
+			}
+
+			actualWorkers := wp.WorkerCount()
+			if tt.expectedWorkers > 0 && actualWorkers != tt.expectedWorkers {
+				t.Errorf("WorkerCount() = %d, want %d", actualWorkers, tt.expectedWorkers)
+			} else if tt.expectedWorkers < 0 && actualWorkers <= 0 {
+				t.Errorf("WorkerCount() = %d, should be positive (default)", actualWorkers)
+			}
+
+			actualQueueSize := wp.QueueSize()
+			if actualQueueSize != tt.expectedQueueSize {
+				t.Errorf("QueueSize() = %d, want %d", actualQueueSize, tt.expectedQueueSize)
+			}
+
+			// Verify the pool is functional
+			var executed bool
+			var wg sync.WaitGroup
+			wg.Add(1)
+
+			err := wp.Submit(func() {
+				defer wg.Done()
+				executed = true
+			})
+
+			if err != nil {
```

Comment:
Use testify? The code base depends on it already. Update AI context to never have to tell it again.
Added benefit apart from consistency is reduction in LoC.